packages/botonic-pulumi/src/pulumi-runner.ts

```diff
@@ -92,9 +96,10 @@ export class PulumiRunner {
   ): Promise<Stack> {
     const projectName = this.projectConfig?.projectName || 'botonic'
     const stackName = this.projectConfig?.stackName || 'full-stack'
+    const prefix = generatePrefix(projectName, stackName)
     const args: InlineProgramArgs = {
       projectName,
-      stackName: `${stackToDeploy}-${stackName}`,
+      stackName: `${prefix}-${stackToDeploy}`,
```

Comment:
Has this `-` anything to do with the `SEPARATOR`? If so, you may want to consider its declaration and export to be reused across files.